app/services/available_accounts_finder.rb

```diff
@@ -1,18 +1,27 @@
 class AvailableAccountsFinder
 
-  def initialize(user, facility, current: nil)
+  def initialize(user, facility, current: nil, current_user: nil)
     @user = user
+    @current_user = current_user
     @facility = facility
     @current_account = current
   end
 
   def accounts
-    accounts = @user.accounts.for_facility(@facility).active
+    accounts = available_accounts
     if @current_account && !accounts.include?(@current_account)
       accounts += [@current_account]
     end
     accounts
   end
   alias to_a accounts
 
+  private
+
+  def available_accounts
+    available_accounts = @user.accounts.for_facility(@facility).active
+    available_accounts &= @current_user.accounts.for_facility(@facility).active if @current_user
+    available_accounts
```

Comment:
At this point, I'd look at moving the code that the adds `@current_account` here as well, since it could be written as a single line. It would leave nothing in `#accounts`… meaning you could rename this method to `#accounts` and have no `private` method at all.